File: Group Anagrams/group anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mp;
        
        for(string str: strs) {
            string key = str;
            sort(key.begin(), key.end());
            if (mp.find(key) == mp.end()) {
                mp[key] = vector<string>();
            }
            mp[key].push_back(str);
        }
        
        vector<vector<string>> ans;
        for(auto it = mp.begin(); it != mp.end(); it++) {
            ans.push_back(it->second);
        }
        return ans;
    }
};

File: Group Anagrams/group anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String str: strs) {
            char[] arr = str.toCharArray();
            Arrays.sort(arr);
            String key = String.valueOf(arr);
            if (!map.containsKey(key)) {
                map.put(key, new ArrayList<>());
            }
            map.get(key).add(str);
        }
        
        List<List<String>> ans = new ArrayList<>();
        
        for(String key: map.keySet()) {
            ans.add(map.get(key));
        }
        
        return ans;
    
    }
}